paystand/paystand-magento2
# PPL-154: Console errors and no redirect after payment completed

JIRA
----
https://paystand.atlassian.net/browse/PPL-154

Description
----
With this PR we are changing the following:
- **Remove checkout.js**: The logic to open checkout changed from having an infinite observer watching whether or not the payment method was selected, to having a button rendered that triggers opening checkout. So the logic is now on paystandmagento-directpost.js
- **Include agreement section**: Agreement section is a Magento 2 core feature that we needed to make our plugin compatible with. Now, when there are terms and conditions available, Paystand's payment method will show them and inherit the requirements to check them or not before proceeding to the actual payment. **Not having this compatibility caused the order not being created (and not being redirected to thank you page) even though payment was successful.**
- **Clean console typeError logs**: This error was caused by the automatic init of checkout with previous flow, now that we are not loading checkout automatically, this error is gone
- **Checkout close button**: Now that we can open checkout 
- **Not injecting js when PS payment method is not enabled**: as pointed by customers, even though the Payment method is not enabled from configuration, we could see the errors in console due to the plugin watching and loading checkout beforehand, since this behavior is gone, now we can successfully disable paystand without impact on magento checkout
 
QA Notes
----
- Go to magento.paystand.biz (it already has this version of Plugin)
- Select your products, go to checkout and test paystand payment method 
- After paying with paystand, you should be redirected to the thank you page and an order should be created.
- After a minute or two, your order should be gone from "pending" to "processing" status, meaning the webhook was processed successfully.
